timemachine/potentials/nonbonded.py

```diff
@@ -119,13 +146,40 @@ def nonbonded_v3(
     dij = np.where(keep_mask, dij, 0)
 
     # funny enough lim_{x->0} erfc(x)/x = 0
-    eij_charge = np.where(keep_mask, qij*erfc(beta*dij)*inv_dij, 0) # zero out diagonals
+    eij_charge = np.where(keep_mask, qij * erfc(beta * dij) * inv_dij, 0)  # zero out diagonals
     if cutoff is not None:
         eij_charge = np.where(dij > cutoff, 0, eij_charge)
 
-    eij_total = (eij_lj*lj_rescale_mask + eij_charge*charge_rescale_mask)
+    eij_total = (eij_lj * lj_rescale_mask + eij_charge * charge_rescale_mask)
+
+    return np.sum(eij_total / 2)
+
+
+def nonbonded_v3_on_specific_pairs(conf, params, box, inds_l, inds_r, beta: float, cutoff: Optional[float] = None):
+    """See nonbonded_v3 docstring for more details"""
+
+    # distances and cutoff
+    dij = distance_on_pairs(conf[inds_l], conf[inds_r], box)
+    if cutoff is None:
+        cutoff = np.inf
+    keep_mask = dij <= cutoff
+
+    def apply_cutoff(x):
```

Comment:
nit: Seems like this nested function is unnecessary since it is calling a single function?

Reply:
Personal style preference, but unnecessary. Can revert.

(I need to do `np.where(keep_mask, x, 0)` a few times in the next couple lines, and it seemed less "noisy" to be calling a function with a single argument. I don't think there's any overhead, since Jax JIT will effectively inline this.)